gramex/data.py

```diff
@@ -1498,6 +1499,69 @@ def _insert_mongodb(url, rows, meta=None, database=None, collection=None, **kwar
     return len(result.inserted_ids)
 
 
+def _filter_influxdb(url, controls, args, org=None, bucket=None, query=None, **kwargs):
+    with _influxdb_client(url, org=org, **kwargs) as db:
+        q = db.query_api()
+        _range = args.pop('range')[0]
```

Comment:
Use _offset and _limit instead of range. Handle time conversion as required.

Reply:
We want support for:

```text
?Continent=Europe ► Continent = Europe
?Continent=Europe&Continent=Asia ► Continent = Europe OR Asia. Multiple values are allowed
?Continent!=Europe ► Continent is NOT Europe
?Continent!=Europe&Continent!=Asia ► Continent is NEITHER Europe NOR Asia
OPTIONAL: ?Shapes ► Shapes is not NULL
OPTIONAL: ?Shapes! ► Shapes is NULL
?c1>=10 ► c1 > 10 (not >= 10)
?c1>~=10 ► c1 >= 10. The ~ acts like an =
?c1<=10 ► c1 < 10 (not <= 10)
?c1<~=10 ► c1 <= 10. The ~ acts like an =
?c1>=10&c1<=20 ► c1 > 10 AND c1 < 20
OPTIONAL: ?Name~=United ► Name matches &_format=html
OPTIONAL: ?Name!~=United ► Name does NOT match United
OPTIONAL: ?Name~=United&Continent=Asia ► Name matches United AND Continent is Asia
```

To control the output, you can use these control arguments:

```text
Limit rows: ?_limit=10 ► show only 10 rows
Offset rows: ?_offset=10&_limit=10 ► show only 10 rows starting, skipping the first 10 rows
OPTIONAL: Sort by columns: ?_sort=Continent&_sort=Name ► sort first by Continent (ascending) then Name (ascending)
OPTIONAL: Sort order: ?_sort=-Continent&_sort=-ID ► sort first by Continent (descending) then ID (descending)
OPTIONAL: Specific columns: ?_c=Continent&_c=Name ► show only the Continent and Names columns
OPTIONAL: Exclude columns: ?_c=-Continent&_c=-Name ► show all columns except the Continent and Names columns
```

Use `_pop_controls()`, `_filter_col()`, maybe `_filter_sort_columns()` from data.py in your code.